### DISCUSSION TITLE
Can I disable code blocks entirely?

### DISCUSSION THREAD

**Discussion by PEZ** [markedjs/marked#2578] [Q&A] [answered]
Hello, I am totally new to this library. Thanks for providing it!

I want this:

```
    # foo
```

To render as:

```
<h1 id="foo">foo</h1>
```

Is it possible? And if so, how would I go about it? I don't need any `code` support what so ever, so if the solution also disables ` ``` ` handling, that is totally fine.

**Comment by UziTech** [comment, marked as answer]
The CommonMark spec states that headings only work [up to 3 spaces](https://spec.commonmark.org/0.30/#example-70). So even if you turn off code blocks the other blocks may not work as expected. I think there are two ways for you to do this:

1. Remove spaces before sending the string to marked than add spaces back if neccessary.
2. Create your own [tokenizer](https://marked.js.org/using_pro#tokenizer) that doesn't follow CommonMark spec.

**Reply by PEZ** [reply]
Thanks! If I start with ignoring the rule of up to 3 spaces (the heading was just an example anyway, I mainly just want 4+ indented text to not be interpreted as code). And I go with option two there. I tried with this tokenizer:

``` js
const tokenizer = {
  code(src) {
    return false;
  }
}
```

Thinking it would disable code blocks. But it doesn't.

``` js
marked.parse('   foo') => "<p>   foo</p>\n"
marked.parse('    foo') => "<pre><code>foo\n</code></pre>\n"
```

What am I not getting here?

(I've tried with the latex codespan example, and that one works.)

**Reply by PEZ** [reply]
OK. So maybe `false` means it should fall back on default. However:

> Returning any other value (including nothing) will prevent fallback behavior.

So trying with returning `null`, but still get the default behaviour.

**Reply by PEZ** [reply]
I'm sorry. I was having other errors when doing the test. I got it to work using:

``` js
const tokenizer = {
  code(src) {
    return null;
  }
}
```

Thanks!

Now next step will be to try get headings and other markup to keep working with more than three spaces...

**Reply by PEZ** [reply]
Update: I never managed to solve this. It seems to be too much against the CommonMark tide. So I switched to a parser that isn't as compliant.

I still love marked. It will be my choice in other projects.
